src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java

```diff
@@ -29,4 +37,42 @@ public CustomerDto getCustomer(final Long customerId) {
                 .orElseThrow(() -> new ResourceNotFoundException("Customer with id=" + customerId + " doesn't exist"));
     }
 
+    @Override
+    public CustomerDto save(CustomerDto customerDto) {
+        Customer customer = Optional.ofNullable(customerDto.getId())
+                .map(item -> {
+                    final Customer existing = customerRepository
+                            .findById(customerDto.getId())
+                            .orElseThrow();
+                    customerMapper.update(customerDto, existing);
+                    return existing;
+                })
+                .orElseGet(() -> customerMapper.map(customerDto));
+
+        final Customer saved = customerRepository.save(customer);
+        userService.save(createAdmin());
+        return customerMapper.map(saved);
+    }
+
+    @Override
+    public List<Customer> saveNewStatus(List<CustomerDto> customerDtoList) {
+        List<Customer> customers = new ArrayList<>();
+        for (CustomerDto customer : customerDtoList) {
+            Customer customerFromDto = customerMapper.map(customer);
+            customerRepository.save(customerFromDto);
+            customers.add(customerFromDto);
+        }
+        return customers;
+    }
+
+    private UserDto createAdmin() {
+        return UserDto.builder()
+                .username("User name")
+                .name("Name")
+                .surname("Surname")
+                .email("email")
```

Comment:
u should set email value sent from frontend

Reply:
It's not obligatory field while creating default admin, am I right? This field can be filled at the time of editing the admin, so I don't have to invent this as default

Reply:
1. according to your code you fill email field in the user table with value 'email' - this creates an inconsistent state of the database, you should not think that someone will replace this 'email' with another really existed;
2. according to the specification Examples of workflows, Scenario1.  "The customer should provide an email address that is entered by the system admin in the form at the customer management page" etc.